src/contract-load.ts

```diff
@@ -32,7 +32,13 @@ export async function loadContract(arweave: Arweave, contractID: string, contrac
     state = contractTX.get('data', { decode: true, string: true });
   }
 
-  const { handler, swGlobal } = createContractExecutionEnvironment(arweave, contractSrc, contractID, contractOwner);
+  const { handler, swGlobal } = createContractExecutionEnvironment(
+    arweave,
+    contractSrc,
+    contractID,
+    contractSrcTXID,
```

Comment:
IMO there's is some serious lack of naming consistency...
`contractSrcTXID`  vs `sourceId` vs `contractSourceId` 🤯  (and `contractID` without `TX` part...)

IMHO naming convention should be:
<subject>TxId

eg:
contractTxId
contractSrcTxId
interactionTxId

etc...

Reply:
In relation to this PR, I'd say your suggestion of `contractSrcTxId` makes sense. The other variable name changes would fit another PR.